doc/specs/stdlib_logger.md

```diff
@@ -34,6 +34,12 @@ The logger variables have the option to:
 * indent subsequent lines of the messages; and
 * format the text to fit within a maximum column width.
 
+While every effort has been made to make the code process and
+asynchronous I/O safe, it is always best to have each process write to
+its own dedicated logger file.
+For thread parallelism (e.g., with OpenMP), it is advised to put the
+logger call in a guarding region (e.g., in an OpenMP critical region).
+
```

Comment:
@milancurcic 
Are you satisfied with this note that covers multi-thread and multi-process programs? If yes, I am ok to merge this PR.

Reply:
Yes, I'm happy with it so I'll go ahead and merge. Thank you all.